utils/preprocess.py

```diff
@@ -1,40 +1,41 @@
-from typing import List
-
-from PIL.Image import Image
+from typing import List, Optional, Tuple, Union
 import numpy as np
 from torch.utils.data import Dataset
 import random
 from tqdm import tqdm
 import torch
+import os
+from pathlib import Path
+from PIL import Image
 
-
-class SIEDataset(Dataset):
-    # load all images at once
-    # we use float32 for the image data type
-    def __init__(self, sand_dust_images: List[Image], ground_truth_images: List[Image], image_names: List[str], w_threshold: int, h_threshold: int):
-        assert len(sand_dust_images) == len(ground_truth_images), "the number of sand dust images and ground truth images must be the same"
+class ImageDataset(Dataset):
+    def __init__(self, sand_dust_images: List[Image.Image], image_names: List[str], ground_truth_images: Optional[List[Image.Image]] = None):
         self.sand_dust_images = sand_dust_images
         self.ground_truth_images = ground_truth_images
         self.image_names = image_names
-        self.w_threshold = w_threshold
-        self.h_threshold = h_threshold
+        self.W_THRESHOLD = 440
+        self.H_THRESHOLD = 330
```

Comment:
Actually these should be class variables

Reply:
I've fixed this issue as well